fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/sink/FlinkSink.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.connector.flink.sink;
+
+import com.alibaba.fluss.annotation.Internal;
+import com.alibaba.fluss.config.Configuration;
+import com.alibaba.fluss.connector.flink.sink.writer.AppendSinkWriter;
+import com.alibaba.fluss.connector.flink.sink.writer.FlinkSinkWriter;
+import com.alibaba.fluss.connector.flink.sink.writer.UpsertSinkWriter;
+import com.alibaba.fluss.metadata.TablePath;
+
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.api.connector.sink2.WriterInitContext;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+/** Flink sink for Fluss. */
+class FlinkSink implements Sink<RowData> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SinkWriterBuilder<? extends FlinkSinkWriter> builder;
+
+    FlinkSink(SinkWriterBuilder<? extends FlinkSinkWriter> builder) {
+        this.builder = builder;
+    }
+
+    @Deprecated
+    @Override
+    public SinkWriter<RowData> createWriter(InitContext context) throws IOException {
+        throw new UnsupportedOperationException(
```

Comment:
We can make `FlinkSinkWriter#initialize` to accept a `SinkWriterMetricGroup` parameter instead of a `WriterInitContext`, then we can also implement this method. I'm afraid some old Flink version calls the deprecated method, and this change can allow us be compatible with the old Flink version.